src/browser/web_storage.rs

```diff
@@ -174,8 +191,14 @@ pub trait WebStorage {
     /// or serialize the value or insert/update the pair.
     ///
     /// [MDN reference](https://developer.mozilla.org/en-US/docs/Web/API/Storage/setItem)
-    fn insert<T: Serialize + ?Sized>(key: impl AsRef<str>, value: &T) -> Result<()> {
-        let value = serde_json::to_string(value).map_err(WebStorageError::SerdeError)?;
+    fn insert<S, T>(key: S, value: &T) -> Result<()>
+    where
+        S: AsRef<str>,
+        T: Serialize,
+    {
+        let js: JsValue = swb::to_value(value)?;
+        let st: JsString = JSON::stringify(&js).map_err(WebStorageError::StringifyError)?;
+        let value: String = st.as_string().ok_or(WebStorageError::ConversionError)?;
```

Comment:
I would rename `js` and `st` to`value`.  All of them represent value, you don't need to use weird names and there is a chance Rust is/will be clever enough to drop old value forms as soon as possible.